sklearn/preprocessing/tests/test_data.py

```diff
@@ -2085,11 +2103,22 @@ def test_power_transformer_strictly_positive_exception():
                          pt.fit, np.zeros(X_2d.shape))
 
     assert_raise_message(ValueError, not_positive_message,
-                         power_transform, np.zeros(X_2d.shape))
+                         power_transform, np.zeros(X_2d.shape), 'box-cox')
+
+    # It should not raise any error for yeo-johnson
```

Comment:
I would do an additional test by parametrizing X_train, X_test with different sign as you did.